pandas/core/indexes/base.py

```diff
@@ -3572,7 +3572,7 @@ def _intersection_via_get_indexer(
         return result
 
     @final
-    def difference(self, other, sort=None):
+    def difference(self, other, sort=False):
```

Comment:
This is an API change so this bug cannot be fixed this way

Reply:
@mroeschke Thanks for the feedback.  Any suggestions on how I should proceed?  Can I add to the body within the `remove_categories` method?

Reply:
I just uploaded a new solution that simply uses an if/else block in the `remove_categories` method to check if ordered has been set to True. This doesn't override any default values in the API so hopefully this will be a better solution.